### ISSUE TITLE
Loss functions

### ISSUE THREAD

**Issue by GoodStuff11** [GoodStuff11/Neural-Network#2]
Convert the loss functions into classes, defining their derivatives.

**Comment by GoodStuff11** [comment]
System has been implemented. More is required:
* cross entropy loss/log loss
* hinge loss
* logistic loss

**Comment by GoodStuff11** [comment]
This class should have a base abstract class called Optimizer and derived classes for each of the algorithms. It should be fully self-contained, not just having the functions
